LEGO1/mxbitmap.cpp

```diff
@@ -0,0 +1,105 @@
+#include "mxbitmap.h"
+#include "mxbool.h"
+#include "mxpalette.h"
+
+// OFFSET: LEGO1 0x100bc980
+MxBitmap::MxBitmap()
+{
+    this->m_info = NULL;
+    this->m_bmiHeader = NULL;
+    this->m_paletteData = NULL;
+    this->m_data = NULL;
+    this->m_unk18 = 0;
+    this->m_palette = NULL;
+}
+
+// OFFSET: LEGO1 0x100bca10
+MxBitmap::~MxBitmap()
+{
+    if (this->m_info != NULL) {
+        free(this->m_info);
+    }
+    if (this->m_data != NULL) {
+        free(this->m_data);
+    }
+    if (this->m_palette != NULL) {
+        // FIXME: Once m_palette/base is understood
+    }
+    // FIXME: Unwind@100bca91. Stack unwinding, I've never done this before.
+}
+
+// OFFSET: LEGO1 0x100bd1c0
+MxPalette *MxBitmap::CreatePalette()
+{
+    // FIXME: This function needs MxPalette to be completed. Also INFERRING usage of MxBool
+    MxPalette *pal;
+    MxPalette *ppal;
+    MxBool success;
+
+    pal = NULL;
+    success = MX_FALSE;
+    if(this->m_unk18 == '\0') {
+        ppal = (MxPalette *)malloc(0x414);
+        pal = NULL;
+        if(ppal != NULL) {
+            // pal = MxPalette::FromBitmapPalette(ppal, this->m_paletteData);
+        }
+    } else {
+        if(this->m_unk18 != '\x01') {
+            goto LAB_100bd253;
+        }
+        //. pal = MxPalette::Clone(this->m_palette);
+    }
+    if(pal != NULL) {
+        success = MX_TRUE;
+    }
+
+// maybe we should rename this ufnction to like 'nullcheck'
+LAB_100bd253:
+    if((!success) && (pal != NULL)) {
+        // (**(pal->base).vtable)(1);
+        pal = NULL;
+    }
+
+    return pal;
+}
+
+// OFFSET: LEGO1 0x100bcd10
+long MxBitmap::Read(char const *filename)
+{
+    HANDLE hObject;
+    int iVar1;
+    long lVar2;
+
+    lVar2 = -1;
+    hObject = CreateFileA(filename,GENERIC_READ,FILE_SHARE_READ,(LPSECURITY_ATTRIBUTES)NULL,OPEN_EXISTING,FILE_ATTRIBUTE_NORMAL,(HANDLE)NULL);
```

Comment:
Formatting + use spaces + name your variables appropriately.
This is c++, no c90 where variables have to be declared at the start of a method.

Reply:
I think lVar2 is just a MxResult, thinking about it.